### PR TITLE
Sample test to launch web test in android emulator

### PR DESCRIPTION
**Pull request by nareshmahendra**
This PR covers :

- Added a new config & caps for android_emulator_chrome

- Added sample test to launch web test in android emulator

Screenshots:

image

image

### CODE REVIEW

File: src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java [hunk 1]
```diff
@@ -18,6 +20,9 @@ public ScreenShotScreenAndroid(Driver driver, Visual visually) {
     @Override
     public ScreenShotScreen takeScreenshot() {
         visually.checkWindow(SCREEN_NAME, "Take Screenshot");
+        driver.getInnerDriver().get("https://github.com/znsio/teswiz");
+        waitFor(3);
```
Comment (raghavgarg1996) [inline src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java:24]: We are not supposed to add wait.

you can use waittillelementis present

File: src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java [hunk 2]
```diff
@@ -18,6 +20,9 @@ public ScreenShotScreenAndroid(Driver driver, Visual visually) {
     @Override
     public ScreenShotScreen takeScreenshot() {
         visually.checkWindow(SCREEN_NAME, "Take Screenshot");
+        driver.getInnerDriver().get("https://github.com/znsio/teswiz");
```
Comment (raghavgarg1996) [inline src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java:23]: Create constant for  this url

File: src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java [hunk 3]
```diff
@@ -18,6 +20,9 @@ public ScreenShotScreenAndroid(Driver driver, Visual visually) {
     @Override
     public ScreenShotScreen takeScreenshot() {
         visually.checkWindow(SCREEN_NAME, "Take Screenshot");
```
Comment (raghavgarg1996) [inline src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java:22]: Add proper description

File: src/test/resources/com/znsio/teswiz/features/mobile-emulation-on-web-browser.feature [hunk 4]
```diff
@@ -7,4 +7,10 @@ Feature: Mobile Emulation on Web browser
...
     And "someone-else" starts "bing-web" on "firefox-tab2"
     When "someone" searches for "bear"
-    Then "someone-else" searches for "tiger"
\ No newline at end of file
+    Then "someone-else" searches for "tiger"
+
+    #  CONFIG=./configs/android_emulator_config.properties TAG="@mobileEmulation and @android-chrome" ./gradlew run
+  @android-chrome @android
+  Scenario: Sample test to launch chrome on android emulator
+    Given "I" starts "chrome-android" on "chrome"
```
Comment (raghavgarg1996) [inline src/test/resources/com/znsio/teswiz/features/mobile-emulation-on-web-browser.feature:15]: It looks very odd to me
you can write something like
Given: I launch chrome browser on my android emulator

File: src/test/resources/com/znsio/teswiz/features/mobile-emulation-on-web-browser.feature [hunk 5]
```diff
@@ -7,4 +7,10 @@ Feature: Mobile Emulation on Web browser
...
+    When "I" searches for "bear"
```
Comment (raghavgarg1996) [inline src/test/resources/com/znsio/teswiz/features/mobile-emulation-on-web-browser.feature:16]: It should be something like
When "I" search for "bear", then search result appears

File: src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java (hunk 1 above)
Comment (nareshmahendra) [reply src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java]: Reverted file

File: src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java (hunk 3 above)
Comment (nareshmahendra) [reply src/test/java/com/znsio/teswiz/screen/android/ScreenShotScreenAndroid.java]: Reverted file

File: src/test/resources/com/znsio/teswiz/features/mobile-emulation-on-web-browser.feature (hunk 5 above)
Comment (nareshmahendra) [reply src/test/resources/com/znsio/teswiz/features/mobile-emulation-on-web-browser.feature]: Reverted file

File: src/test/java/com/znsio/teswiz/screen/android/googlesearch/GoogleSearchLandingScreenAndroid.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        this.visually = visually;
+        long threadId = Thread.currentThread().getId();
+        context = Runner.getTestExecutionContext(threadId);
+        driver.getInnerDriver().get(URL);
+        driver.waitTillElementIsPresent(SEARCH_INPUT);
+    }
+
+    @Override
+    public GoogleSearchResultsScreen searchFor(String searchText) {
+        driver.findElement(SEARCH_INPUT).sendKeys(searchText, Keys.ENTER);
```
Comment (anandbagmar) [inline src/test/java/com/znsio/teswiz/screen/android/googlesearch/GoogleSearchLandingScreenAndroid.java:34]: visual validations missing

File: src/test/java/com/znsio/teswiz/businessLayer/googlesearch/GoogleSearchBL.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        long threadId = Thread.currentThread().getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = userPersona;
+        this.currentPlatform = forPlatform;
+        Runner.setCurrentDriverForUser(userPersona, forPlatform, context);
+    }
+
+    public SearchResultsBL searchFor(String searchText) {
+        GoogleSearchLandingScreen.get().searchFor(searchText);
```
Comment (anandbagmar) [inline src/test/java/com/znsio/teswiz/businessLayer/googlesearch/GoogleSearchBL.java:26]: assertion missing

File: src/test/java/com/znsio/teswiz/screen/android/googlesearch/GoogleSearchLandingScreenAndroid.java (hunk 6 above)
Comment (nareshmahendra) [reply src/test/java/com/znsio/teswiz/screen/android/googlesearch/GoogleSearchLandingScreenAndroid.java:35]: Added

File: src/test/java/com/znsio/teswiz/businessLayer/googlesearch/GoogleSearchBL.java (hunk 7 above)
Comment (nareshmahendra) [reply src/test/java/com/znsio/teswiz/businessLayer/googlesearch/GoogleSearchBL.java]: Added assertions

**Pull request merged**